### PR TITLE
UCP/AM: Fix stub ep multi-bcopy flow

### PR DESCRIPTION
**Pull request by brminich**
## What
Fix the case when send is started as bcopy-multi, but just one packet was sent (due to different bcopy limits in stub and regular eps)

### CODE REVIEW

**Comment by brminich** [comment]
bot:pipe:retest

File: src/ucp/core/ucp_am.c [hunk 1]
```diff
@@ -140,26 +140,26 @@ ucp_am_bcopy_pack_args_single_reply(void *dest, void *arg)
     return hdr_size + length;
 }
 
-static size_t 
+static size_t
```
Comment (yosefe) [inline src/ucp/core/ucp_am.c:143]: pls add AM API to test_ucp_sockaddr protocols

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by brminich** [comment]
waiting for #5074 to update this PR with tests and more fixes.
the fix in #5074 is unrelated to stub eps, but quite important to be in a separate PR

**Comment by brminich** [comment]
@yosefe, added test

File: src/ucp/core/ucp_am.c [hunk 2]
```diff
@@ -205,10 +205,10 @@ static ucs_status_t ucp_am_send_short(ucp_ep_h ep, uint16_t id,
 static ucs_status_t ucp_am_contig_short(uct_pending_req_t *self)
 {
     ucp_request_t *req   = ucs_container_of(self, ucp_request_t, send.uct);
-    ucs_status_t  status = ucp_am_send_short(req->send.ep, 
-                                             req->send.msg_proto.am.am_id, 
-                                             req->send.buffer, 
-                                             req->send.length);
+    ucp_ep_t *ep         = req->send.ep;
+    req->send.lane       = ucp_ep_get_am_lane(ep);
```
Comment (yosefe) [inline src/ucp/core/ucp_am.c:209]: looks like mixed declaration/code

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 3]
```diff
@@ -951,6 +951,30 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
         ASSERT_UCS_OK(status);
     }
 
+    void test_am_send_recv(size_t size)
+    {
+        std::string sb(size, 'x');
+
+        bool am_received = false;
+        ucp_worker_set_am_handler(receiver().worker(), 0,
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:959]: where we remove it?

Comment (brminich) [reply test/gtest/ucp/test_ucp_sockaddr.cc:959]: what API do you mean?

Comment (yosefe) [reply test/gtest/ucp/test_ucp_sockaddr.cc:959]: ucp_worker_set_am_handler
```
 * @param [in]  cb          Active Message callback. NULL to clear.
```

Comment (brminich) [reply test/gtest/ucp/test_ucp_sockaddr.cc:959]: is it required by API? What is the reason to clear in this test?

Comment (yosefe) [reply test/gtest/ucp/test_ucp_sockaddr.cc:959]: more for a good practice to release what is allocated

**Comment by brminich** [comment]
@yosefe, ok to squash?

**Pull request merged**